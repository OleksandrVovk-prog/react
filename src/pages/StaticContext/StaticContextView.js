import { useContext } from 'react';
import { Context } from '../../context/pages/StaticContext';
import styles from './sass/StaticContext.module.scss';

export default function StaticContextView() {
  const { title } = useContext(Context);

  return (
    <div className={styles.staticContext}>
      <div className={styles.events}>
        <p>{title}</p>
      </div>
    </div>
  );
}
