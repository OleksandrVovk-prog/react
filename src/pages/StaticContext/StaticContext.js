import StaticContextView from './StaticContextView';
import { Context, data } from '../../context/pages/StaticContext';

export default function StaticContext() {
  return (
    <Context.Provider value={data}>
      <StaticContextView />
    </Context.Provider>
  );
}
