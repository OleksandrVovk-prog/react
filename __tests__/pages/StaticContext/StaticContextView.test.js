import React from 'react';
import { render } from '@testing-library/react';
import StaticContextView from '../../../src/pages/StaticContext/StaticContextView';
import { data } from '../../../src/context/pages/StaticContext';

describe('<StaticContextView />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = render(<StaticContextView />);
  });
  test('should render StaticContextView component', () => {
    const { asFragment } = wrapper;
    expect(asFragment(wrapper)).toMatchSnapshot();
  });
  test('should contain context data', () => {
    const { getByText } = wrapper;
    const Element = getByText(data.title);
    expect(Element).toBeInTheDocument();
  });
});
