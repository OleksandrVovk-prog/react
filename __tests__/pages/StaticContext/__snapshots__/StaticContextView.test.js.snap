// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StaticContextView /> should render StaticContextView component 1`] = `
<DocumentFragment>
  <div
    class="staticContext"
  >
    <div
      class="events"
    >
      <p>
        In vino veritas, in aqua sanitas
      </p>
    </div>
  </div>
</DocumentFragment>
`;
